torch/distributed/pipeline/sync/pipe.py

```diff
@@ -306,21 +307,24 @@ def _ensure_copy_streams(self) -> List[List[AbstractStream]]:
 
         return self._copy_streams
 
-    def forward(self, input: TensorOrTensors) -> RRef[TensorOrTensors]:  # type: ignore
-        """:class:`Pipe` is a fairly transparent module wrapper. It doesn't
+    def forward(self, input) -> RRef:  # type: ignore
+        """
+        Processes a single input mini-batch through the pipe and returns an
+        :class:`~torch.distributed.rpc.RRef` pointing to the output.
+        :class:`Pipe` is a fairly transparent module wrapper. It doesn't
         modify the input and output signature of the underlying module. But
         there's type restriction. Input and output have to be a
         :class:`~torch.Tensor` or a sequence of tensors. This restriction is
         applied at partition boundaries too.
```

Comment:
Any reason for this limitation? Are we going to drop this in the future?

Reply:
Its tricky to split an arbitrary python object into multiple microbatches, that is the primary motivation of this limitation.